### PR TITLE
Polkadot v0.9.38

### LINKED ISSUE #73: [Tech]: Disable any direct access to the XCM Executor via extrisics
### Problem statement

Since our parachain does not have tokens we should ensure that there is no access to xcm executor directly through extrinsics

### Description

_No response_

### Definition of Done

Check that no pallet like orml_xtokens and pallet_xcm give such access

### Requirements

_No response_

### LINKED ISSUE #82: sora2-parachain uses v0.9.38

### PR DESCRIPTION
**Pull request by N1ghtStorm**

### CODE REVIEW

File: runtime/src/trader.rs [hunk 1]
```diff
@@ -40,7 +40,7 @@ pub struct ParachainTrader {
 impl WeightTrader for ParachainTrader {
 	fn new() -> Self {
 		log::trace!(target: "xcm::weight", "creating new WeightTrader instance");
-		Self { weight: 0, multi_location: None }
+		Self { weight: XcmWeight::from_ref_time(0), multi_location: None }
```
Comment (wer1st) [inline runtime/src/trader.rs]: Is there `XcmWeight::zero()`?
If yes, it is better to use

File: runtime/src/trader.rs [hunk 2]
```diff
@@ -70,10 +71,10 @@ impl WeightTrader for ParachainTrader {
 		match &self.multi_location {
 			None => None,
 			Some(ml) => {
-				if weight == 0 {
+				if weight.ref_time() == 0 {
```
Comment (wer1st) [inline runtime/src/trader.rs]: Is there method `weight.is_zero()` ?

File: runtime/src/xcm_config.rs [hunk 3]
```diff
@@ -289,9 +242,73 @@ impl orml_xtokens::Config for Runtime {
...
 	type MultiLocationsFilter = Everything;
 	type ReserveProvider = AbsoluteReserveProvider;
+	type UniversalLocation = UniversalLocation;
+}
+
+// The pallet will be disabled for extarnal calls
+pub struct PalletXCMWeightInfo;
+impl pallet_xcm::WeightInfo for PalletXCMWeightInfo {
+	fn send() -> XcmWeight {
+		XcmWeight::from_ref_time(0)
```
Comment (wer1st) [inline runtime/src/xcm_config.rs]: `XcmWeight::zero()` if this method exists

Comment (N1ghtStorm) [reply runtime/src/xcm_config.rs]: Done

File: runtime/src/trader.rs (hunk 2 above)
Comment (N1ghtStorm) [reply runtime/src/trader.rs]: yes, there is, Done

**Review by wer1st — APPROVED** [review]

File: runtime/src/xcm_config.rs [hunk 4]
```diff
@@ -47,17 +46,19 @@ use sp_core::Get;
...
+	RelayChainAsNative, SiblingParachainAsNative, SiblingParachainConvertsVia,
 	SignedAccountId32AsNative, SignedToAccountId32, SovereignSignedViaLocation, TakeWeightCredit,
 };
-use xcm_executor::{traits::ShouldExecute, XcmExecutor};
+use xcm_executor::XcmExecutor;
 
 parameter_types! {
 	pub const RelayLocation: MultiLocation = MultiLocation::parent();
-	pub const RelayNetwork: NetworkId = NetworkId::Any;
+	pub const RelayNetwork: NetworkId = NetworkId::Rococo;
```
Comment (vovac12) [inline runtime/src/xcm_config.rs:57]: I guess we will need to change this value for Kusama and Polkadot?

**Review by vovac12 — APPROVED** [review]

File: runtime/src/xcm_config.rs (hunk 4 above)
Comment (N1ghtStorm) [reply runtime/src/xcm_config.rs:57]: > I guess we will need to change this value for Kusama and Polkadot?

Sure

**Pull request merged**